### ISSUE TITLE
Fix review 1 issues

### ISSUE THREAD

**Issue by KumarHE** [KumarHE/Gen-AI#1]
Reviewer Note

❌❌❌
Please note that as per the project instructions, you need to evaluate the base model.
image.png

Please have a look at the Project Guide in [the classroom(opens in a new tab)](https://learn.udacity.com/nanodegrees/nd608/parts/cd13303/lessons/5218a064-df18-4aac-8b2b-772d57545dbe/concepts/e0f0b83b-3f0a-4a25-be14-0939910eff91?_gl=1*1u9nlsd*_gcl_au*MTAyNTkzNTIyNS4xNzI3MDUzMjE4*_ga*MTIzOTg3Njc0OS4xNzI3MDUzMzI1*_ga_CF22GKVCFK*MTczMTYwODE0Ny40Ni4wLjE3MzE2MDgxNDcuNjAuMC4w&lesson_tab=lesson). It will help you a lot in easily understanding the structure of the project and clarifying with examples what you should do.

We have noticed that you created a PEFT model, trained it then evaluated it. But, you haven't evaluated the base model. You need to evaluate the base model before creating the PEFT one.

In short, you will have to create a trainer that will be used for model evaluation, and a compute_metrics function to calculate the model evaluation metric (Accuracy, F1, Recall, or any other metric). So your code may look like this:

training_args = TrainingArguments(
    output_dir='./results/foundation_model',
    evaluation_strategy="epoch",
    per_device_eval_batch_size=16,
    logging_dir='./logs',
)

trainer = Trainer(
    model=model,
    args=training_args,
    eval_dataset=tokenized_datasets['test'],  # Use the validation split for evaluation
    tokenizer=tokenizer,
    data_collator=DataCollatorWithPadding(tokenizer=tokenizer),
    compute_metrics=compute_metrics,
)
Pass this function in the Trainer as compute_metrics=compute_metrics

import numpy as np
from datasets import load_metric

# Define the accuracy metric
accuracy_metric = load_metric("accuracy")

# Compute metrics function
def compute_metrics(eval_pred):
    # Extract predictions and labels
    logits, labels = eval_pred
    predictions = np.argmax(logits, axis=-1)  # Convert logits to predicted class indices
    
    # Compute accuracy
    accuracy = accuracy_metric.compute(predictions=predictions, references=labels)
    
    return accuracy
and then you can evaluate the model using the .evaluate method. Your code may look something like that: print( trainer.evaluate()).

Note: The code snippets here are only a guide. Feel free to adapt them to work with your own variables.

At least one classification metric is calculated using the dataset and pretrained model

================
Perform Inference Using the Fine-Tuned Model:

Reviewer Note

You successfully used AutoPeftModelForSequenceClassification to load the previously saved model. ✔️✔️

💭💭However, We need to pay attention to the description of the task. As seen in the provided screenshot,

image.png

you need to implement the following ToDO instructions to pass this criterion:

Load the saved PEFT model weights, which you have done successfully. ✔️✔️
Evaluate the performance of the trained PEFT model. ❌
Be sure to compare the results to the results you got before fine-tuning.❌
To do that let me share some tips with you showing how these tasks might be achieved.
In your case, and after that you have uploaded the previously saved model, you will need to pass it as the new model parameter for the previously created trainer
trainer.model = inference_model

Then now you can again evaluate the uploaded model by using the .evaluate method as follows trainer.evaluate().
See, what are we doing here is so simple. We are saving the PEFT model and uploading it again for evaluation. Piece of cake. 😀

Then you would need at the end to see an accuracy of the loaded model similar to it before saving, which was, in your case, 98%. Now you can compare this result with the accuracy of the pre-trained model before fine-tuning.
Repeats the earlier evaluation process (same metric(s) and dataset) to compare the fine-tuned version to the original version of the model
